docs/concepts/results.md

```diff
@@ -193,3 +193,191 @@ def my_flow():
 
 my_flow()
 ```
+
+## Persisting results
+
+The Prefect API does not store your results [except in special cases](#storage-of-results-in-prefect). Instead, the result is _persisted_ to a storage location in your infrastructure and Prefect stores a _reference_ to the result.
+
+The following Prefect features require results to be persisted:
+
+- Task cache keys
+- Flow run retries
```

Comment:
I think this won't be a universally true statement -- a flow run composed of tasks that don't exchange data should be able to retry regardless of this setting. Recognizing that this is a pedantic point, perhaps `- Retrying flows that exchange data` would be more specific?

Reply:
That's true, but as-is we're going to turn on results if you have flow run retries enabled whether or not your tasks exchange data. We don't know if your task's data will be used downstream when it runs and we need to decide if it should be persisted or not.

Reply:
Oh sweet 👍